### PR TITLE
Reduce complexity of panels endpoint by moving some code to the panels controllers

### PR DESCRIPTION
**Pull request by northwestwitch**
Codefactor check is flagging the current  `panels` endpoint as too complex and this function can be easily simplified!

It's just a refactor PR, with no code added or removed.

**How to test**:
1. Start a local instance of Scout and try to add and modify genes for a gene panel

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/panels/views.py [hunk 1]
```diff
@@ -19,64 +19,9 @@
 @panels_bp.route("/panels", methods=["GET", "POST"])
 @templated("panels/panels.html")
 def panels():
-    """Show all panels for a case."""
+    """Show all panels for a user"""
     if request.method == "POST":
-        # update an existing panel
```
Comment (northwestwitch) [inline scout/server/blueprints/panels/views.py:24]: This part is not changed, just moved to the controllers!

File: scout/server/blueprints/panels/views.py [hunk 2]
```diff
@@ -19,64 +19,9 @@
 @panels_bp.route("/panels", methods=["GET", "POST"])
 @templated("panels/panels.html")
 def panels():
-    """Show all panels for a case."""
+    """Show all panels for a user"""
```
Comment (dnil) [inline scout/server/blueprints/panels/views.py:22]: 👍

File: scout/server/blueprints/panels/controllers.py [hunk 3]
```diff
@@ -13,6 +14,83 @@
 INHERITANCE_MODELS = ["ar", "ad", "mt", "xr", "xd", "x", "y"]
 
 
+def shall_display_panel(panel_obj, user):
+    """Check if panel shall be displayed based on display status and user previleges."""
+    is_visible = not panel_obj.get("hidden", False)
+    return is_visible or panel_write_granted(panel_obj, user)
+
+
+def panel_write_granted(panel_obj, user):
```
Comment (dnil) [inline scout/server/blueprints/panels/controllers.py:23]: Indeed - for want of a panel class, controllers is a better home than views.

**Review by dnil — APPROVED** [review]
Looks great - and we'll have to test this more anyway.

**Pull request merged**